component/azstorage/datalake.go

```diff
@@ -624,3 +624,8 @@ func (dl *Datalake) SetFilter(filter string) error {
 
 	return dl.BlockBlob.SetFilter(filter)
 }
+
+// SetMetadata : Set metadata property of path
+func (dl *Datalake) SetMetadata(name string, attr *internal.ObjAttr) (err error) {
```

Comment:
We generally do not rely on this kind of interface where we send attr and assume its metadata is correct, rather explicitly send the metadata.